src/commonTest/kotlin/exchange.dydx.abacus/app/manager/v2/V4RestrictionsTests.kt

```diff
@@ -94,6 +96,88 @@ class V4RestrictionsTests : NetworkTests() {
         )
     }
 
+    @Test
+    fun testGeoEndpointHandling() {
+        reset()
+
+        testChain!!.signCompliancePayload = """
+        {
+            "signedMessage": "1",
+            "publicKey": "1",
+            "timestamp": "2024-05-14T20:40:00.415Z"
+        }
+        """.trimIndent()
+
+        testRest?.setResponse(
+            "https://indexer.v4staging.dydx.exchange/v4/compliance/geoblock",
+            """
+                {
+                    "status": "CLOSE_ONLY",
+                    "reason": null,
+                    "updatedAt": "2024-05-14T20:40:00.415Z"
+                }
+            """.trimIndent(),
+        )
+        testRest?.setResponse(
+            "https://indexer.v4staging.dydx.exchange/v4/compliance/screen/$testCosmoAddress",
+            """
+                {
+                    "status": "CLOSE_ONLY",
+                    "reason": null,
+                    "updatedAt": "2024-05-14T20:40:00.415Z"
+                }
+            """.trimIndent(),
+        )
+
+        setStateMachineConnected(stateManager)
+        stateManager.setAddresses(null, testCosmoAddress)
+
+        testRest?.setResponse(
+            "https://indexer.v4staging.dydx.exchange/v4/compliance/screen/$testCosmoAddress",
+            """
+                {
+                    "status": "CLOSE_ONLY",
+                    "reason": null,
+                    "updatedAt": "2024-05-14T20:40:00.415Z"
+                }
+            """.trimIndent(),
+        )
+
+        testRest?.setResponse(
+            "https://indexer.v4staging.dydx.exchange/v4/compliance/geoblock",
+            """
+                {
+                    "status": "CLOSE_ONLY",
+                    "reason": null,
+                    "updatedAt": "2024-05-14T20:40:00.415Z"
+                }
+            """.trimIndent(),
+        )
+
+        stateManager.triggerCompliance(ComplianceAction.CONNECT) { successful, error, data ->
+            print("")
+        }
```

Comment:
@rosepuppy / @johnqh can you please see [this comment](https://github.com/dydxprotocol/v4-abacus/pull/333/files#r1587797289) in a previous PR. Is there a reason that `triggerCompliance` is with a callback? It doesn't seem to have async code:

```
    internal open fun triggerCompliance(action: ComplianceAction, callback: TransactionCallback) {
        accountAddress?.let {
            if (compliance.status != ComplianceStatus.UNKNOWN) {
                updateCompliance(DydxAddress(it), compliance.status, action)
                callback(true, null, null) //this is not an asynchronous usage of a callback
            }
        }
        callback(false, V4TransactionErrors.error(null, "No account address"), null)
    }
```

Reply:
hmmm good point, this callback doesn't make sense here. I guess the call back should actually be passed into from updateCompliance potentially 🤔

Reply:
i will leave out of this PR for now since it's out of the OG scope